storage/remote/client.go

```diff
@@ -207,6 +207,9 @@ func (c *Client) Store(ctx context.Context, req []byte) error {
 	httpReq.Header.Set("Content-Type", "application/x-protobuf")
 	httpReq.Header.Set("User-Agent", UserAgent)
 	httpReq.Header.Set("X-Prometheus-Remote-Write-Version", "0.1.0")
+	if attempt > 0 {
+		httpReq.Header.Set("X-Retry-Count", strconv.Itoa(attempt))
```

Comment:
AFAICT this isn't an official HTTP spec header right? If not, what do you think of calling it `X-Retry-Attempt`? To me, `count` here implies the remaining count of retries before we would give up. 

Lets make sure we document the new retry options as part of the 1.1 spec. We should probably also define the header names via constants at some point.